src/main/software/amazon/event/ruler/input/DefaultParser.java

```diff
@@ -12,7 +12,8 @@
  * rule's value will be treated by their byte representation, but certain characters, such as for wildcards or regexes,
  * need to be represented differently so the Machine understands their special meaning.
  */
-public class Parser {
+@SuppressWarnings("UseOfConcreteClass")
+public class DefaultParser implements MatchTypeParser, ByteParser {
```

Comment:
I don't think the renaming is necessary but if you are doing so, please check if there's any documentation (in code or Readme) that need to be updated as well. I've found IDEs to not be able to pick them all.